donut/modules/uploads/routes.py

```diff
@@ -0,0 +1,105 @@
+import flask
+import json
+import os
+from werkzeug import secure_filename
+from donut.modules.uploads import blueprint, helpers
+from donut.resources import Permissions
+from donut.auth_utils import check_permission
+
+
+@blueprint.route('/lib/<path:url>')
+def display(url):
+    '''
+    Displays the webpages that have been created by users.
+    '''
+    page = helpers.read_page(url.replace(' ', '_'))
```

Comment:
2 comments.

One, it seems like [http://127.0.0.1:9000/lib/BoC_Bylaws](http://127.0.0.1:9000/lib/BoC_Bylaws) and [http://127.0.0.1:9000/lib/BoC Bylaws](http://127.0.0.1:9000/lib/BoC Bylaws) and [http://127.0.0.1:9000/lib/BoC%20Bylaws](http://127.0.0.1:9000/lib/BoC%20Bylaws) all go to the same page because of this replace. Hopefully this is the intended behavior.

Two, you should better handle it if a page does not exist (i.e. has not been created). Right now, loading a page like [http://127.0.0.1:9000/lib/BoCCoB](http://127.0.0.1:9000/lib/BoCCoB) gives a FileNotFound error on the server rather than displaying a Page not found error on the website itself (which for some reason is handled correctly when trying to access uploaded files that were not created).